zio-http/src/main/scala/zhttp/http/middleware/HttpMiddleware.scala

```diff
@@ -52,6 +52,23 @@ object HttpMiddleware {
 
   type RequestP[+A] = (Method, URL, List[Header]) => A
 
+  def ifThenElse[R, E](cond: RequestP[Boolean])(left: HttpApp[R, E], right: HttpApp[R, E]): HttpMiddleware[R, E] =
+    HttpMiddleware.fromMiddlewareFunction((m, u, h) =>
```

Comment:
We can add a fromHttpFunction for simplicity.

Reply:
I don't understand. what you want. What will be the  signature of `fromHttpFunction`

Reply:
```scala
def fromHttpFunction[R, E](f: RequestP[HttpApp[R, E]]): HttpMiddleware[R, E] = ???
def fromHttpFunctionM[R, E](f: RequestP[ZIO[R, E, HttpApp[R, E]]]): HttpMiddleware[R, E] = ???
```

Reply:
Added those new methods